### DISCUSSION TITLE
qBittorrent login failed with code 404

### DISCUSSION THREAD

**Discussion by Kopernikus1979** [cross-seed/cross-seed#392] [Q&A] [answered]
Hi,

Got this error when starting cross-seed:

```
info: Validating your configuration...
error: qBittorrent login failed with code 404
```

Running qBittorrent 4.5.2 and Cross-seed 4.4.1 on Unraid 6.11.5

According to the qBittorent API manual error code 404: "Torrent hash was not found"

**Comment by bakerboy448** [comment]
Your configuration is incorrect and qbit does not exist or is not accessible to xseed at the url/port you've entered

**Comment by Kopernikus1979** [comment]
Qbittorent is working also is qbitmanage

**Comment by bakerboy448** [comment]
Well a single line error saying nothing is there doesn't help....you've provided no information at all. 

- are you missing a base url?
- can you curl qbit from within the cross seed container?
- Is your ip/host and port accurate?
- your understanding of qbit's 404 error is wrong and the login api route has nothing to do with getting torrents as you said

**Comment by Kopernikus1979** [comment]
> Well a single line error saying nothing is there doesn't help....you've provided no information at all.
> 
>     * are you missing a base url?

No: qbittorrent.mydomain.com
> 
>     * can you curl qbit from within the cross seed container?

Yes working fine

> 
>     * Is your ip/host and port accurate?

Static ip and host and port ofcourse

> 
>     * your understanding of qbit's 404 error is wrong and the login api route has nothing to do with getting torrents as you said

I understand, anything alse could be the cause?

**Comment by Kopernikus1979** [comment]
I did some more testing and it seems the cross-seed container is not able to contact qbittorrent and the torznab prowlarr links, tried custom network, bridge and my docker VLAN, all other containers like qbit manage/sonarr/radarr/prowlarr etc can reach qbittorent without a problem, in cross-seed logs I have the error "failed, reason: getaddrinfo EAI_AGAIN" seems to be related to DNS, any idea?

**Comment by bakerboy448** [comment]
sounds like basic docker dns issues or docker network configuration issues; nothing specific to cross-seed

**Comment by mmgoodnow** [comment]
getaddrinfo EAI_AGAIN is not a 404 so you're experiencing at least 2 distinct issues.

**Comment by mmgoodnow** [comment]
https://stackoverflow.com/questions/40182121/whats-the-cause-of-the-error-getaddrinfo-eai-again

**Comment by Kopernikus1979** [comment]
Hi,

DNS problems should be solved now, all is reachable.
The torznab links are also woking when setting "action" to save I see cross-seed searching on all my indexers.
However connection to qbittorrent still gives me error 404... in the container I can do a ping to it and ir responds.
Can it be a setting in qBittorrent or something I can try?

Now running in "save" mode instead of "inject" also working but rather use it with inject

Thx

**Comment by mmgoodnow** [comment]
what are you putting as your qbittorrent url? 404 means it's most likely a base path issue

**Comment by ghost** [comment, marked as answer]
I had same issue - cross-seed reports 404 but curl didn't have any issues loading anything. I looked at the code and I resolved it by removing the trailing slash from qBittorrent URL. Cross-seed will append `/api/v2` to the URL you enter in the config.

IE instead of `https://user:password@qbittorrent.example.org/`, I needed to use `https://user:password@qbittorrent.example.org`.

I think it could be useful to:
1. Display the URL cross-seed is trying to use in the error message/log,
2. Remove the trailing slash from the URL.
